src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +160,515 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  file_system::CreateDirectoryRecursively(directory);
+  file_system::CreateFile(full_file_path);
+  req->Run();
+  file_system::RemoveDirectory(app_storage_folder);
+
+  const std::string expect_sync_file_name =
+      file_system::ConvertPathForURL(full_file_path);
+  const std::string actual_sync_file_name =
+      (*result)[am::strings::msg_params][am::strings::sync_file_name]
+               [am::strings::value].asString();
+  EXPECT_EQ(expect_sync_file_name, actual_sync_file_name);
+
+  const uint32_t expect_app_id = app_id;
+  const uint32_t actual_app_id = static_cast<uint32_t>(
+      (*result)[am::strings::msg_params][am::strings::app_id].asInt());
+  EXPECT_EQ(expect_app_id, actual_app_id);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentApp_ApplicationNotRegistered) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest,
+       Run_SyncFileNameWithForbiddenSymbols_InvalidData) {
+  (*msg_vr)[am::strings::msg_params][am::strings::sync_file_name] =
+      "sync_file/";
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::INVALID_DATA;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentFile_InvalidData) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
```

Comment:
@BSolonenko this variable is used in many places. Good candidate to move it to namespace as a constant

Reply:
@AKalinich-Luxoft  Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/beb0fd60bd310bdbc5a04821177a9005cb5c0fd9